catalog/resource_sql_table.go

```diff
@@ -406,6 +430,23 @@ func ResourceSqlTable() common.Resource {
 	return common.Resource{
 		Schema: tableSchema,
 		CustomizeDiff: func(ctx context.Context, d *schema.ResourceDiff) error {
+			if d.HasChange("column") {
+				old, new := d.GetChange("column")
+				oldCols := old.([]interface{})
+				newCols := new.([]interface{})
+
+				// Only handling same number of columns for now, will address different number of columns as a follow-up.
+				if len(oldCols) == len(newCols) {
+					for i, oldCol := range oldCols {
+						oldColMap := oldCol.(map[string]interface{})
+						newColMap := newCols[i].(map[string]interface{})
+
+						if oldColMap["type"] != newColMap["type"] {
+							return fmt.Errorf("changing the 'type' of an existing column is not allowed, please apply twice by dropping and adding a new column")
```

Comment:
This recommendation will cause the table to be dropped and recreated today. I would leave out the recommendation for now until we actually support removing and adding columns, and just say that changing the type is not supported.